app/components/Views/SDK/SDKDisconnectModal/SDKDisconnectModal.tsx

```diff
@@ -87,54 +87,48 @@ const SDKDisconnectModal = ({ route }: SDKDisconnectModalProps) => {
   }, [channelId, account]);
 
   const onConfirm = async () => {
-    if (account && channelId) {
-      removePermittedAccounts(channelId, [toHex(account)]);
-    } else if (!account && channelId) {
-      SDKConnect.getInstance().removeChannel({
-        channelId,
-        sendTerminate: true,
-      });
-    }
-
-    DevLogger.log(
-      `OnConfirm: accountsLength=${accountsLength} channelId: ${channelId}, account: ${account}`,
-    );
-    if (account && accountsLength && accountsLength <= 1 && channelId) {
-      SDKConnect.getInstance().removeChannel({
-        channelId,
-        sendTerminate: true,
-      });
-    } else if (!account && !channelId) {
-      SDKConnect.getInstance().removeAll();
-    }
-
-    navigate(Routes.SETTINGS.SDK_SESSIONS_MANAGER, { trigger: Date.now() });
-  };
-
-  const onConfirmV2 = async () => {
     try {
-      // Case: Disconnect a single account from a V2 session.
-      // This is a pure permission management action.
-      if (account && channelId) {
-        removePermittedAccounts(channelId, [toHex(account)]);
+      const isGlobalDisconnect = !account && !channelId; // Disconnect all sessions.
+      const isSessionDisconnect = !account && channelId; // Disconnect a specific session.
+      const isAccountDisconnect = account && channelId; // Disconnect a specific account under a session.
+
+      if (isGlobalDisconnect) {
+        // Disconnect all V1 sessions.
+        SDKConnect.getInstance().removeAll();
+        // Disconnect all V2 sessions.
+        const v2ConnectionIds = Object.keys(v2Connections || {});
+        const promises = v2ConnectionIds.map((connId) =>
+          SDKConnectV2.disconnect(connId),
+        );
+        await Promise.all(promises);
+      } else if (isSessionDisconnect) {
+        if (isV2) {
+          await SDKConnectV2.disconnect(channelId as string);
+        } else {
+          SDKConnect.getInstance().removeChannel({
+            channelId: channelId as string,
+            sendTerminate: true,
+          });
+        }
+      } else if (isAccountDisconnect) {
+        // This is a pure permission management action.
+        removePermittedAccounts(channelId as string, [
+          toHex(account as string),
+        ]);
```

Comment:
```suggestion
        removePermittedAccounts(channelId as string, [
          account,
        ]);
```
`removePermittedAccounts` takes in string[], so we don't need `toHex`, or for that matter even typecast `account` since it's already a `string` type.

This should fix the [bug discussed](https://www.notion.so/metamask-consensys/SDK-Phase-1-End-game-276f86d67d688061a461ee0cbaf37649?p=27ef86d67d688050b1fedb18b0011feb&pm=s) regarding Solana accounts being passed in here.